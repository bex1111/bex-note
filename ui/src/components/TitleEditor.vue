<script setup>

import {toRef, watch} from "vue";


const title = defineModel()
const props = defineProps({
  selectedTitle: String
})

const selectedTitleRef = toRef(props, 'selectedTitle');

watch(selectedTitleRef, (newVal) => {
  if (newVal) {
    title.value = newVal;
  }
})

</script>

<template>
  <prime-fluid style=" margin-top: 0.75rem;">
    <prime-float-label variant="in">
      <label for="in_label">Give a title, like: Folder/Awesome title, Awesome title)</label>
      <prime-input id="in_label" v-model="title" type="text"/>
    </prime-float-label>
  </prime-fluid>
</template>

<style scoped>

</style>