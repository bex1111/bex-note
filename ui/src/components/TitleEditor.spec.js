import {mount} from '@vue/test-utils';
import {vi} from 'vitest';
import {nextTick} from 'vue';
import TitleEditor from './TitleEditor.vue';

describe('TitleEditor.vue', () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });

    it('matches snapshot', () => {
        const wrapper = mount(TitleEditor, {
            props: {
                selectedTitle: 'test-title',
                modelValue: 'initial title'
            },
            global: {
                stubs: {
                    'prime-fluid': {
                        template: '<div class="prime-fluid-stub"><slot></slot></div>'
                    },
                    'prime-float-label': {
                        template: '<div class="prime-float-label-stub"><slot></slot></div>'
                    },
                    'prime-input': true
                }
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    });

    describe('watcher for selectedTitle', () => {

        const createWrapper = (props) => {
            return mount(TitleEditor, {
                props,
                global: {
                    stubs: {
                        'prime-fluid': true,
                        'prime-float-label': true,
                        'prime-input': true
                    }
                }
            });
        }

        it('updates title when selectedTitle prop changes to a non-empty value', async () => {
            const wrapper = createWrapper({
                selectedTitle: '',
                modelValue: 'initial title'
            });

            await wrapper.setProps({selectedTitle: 'new-selected-title'});
            await nextTick();

            expect(wrapper.emitted('update:modelValue')).toBeTruthy();
            expect(wrapper.emitted('update:modelValue')[0]).toEqual(['new-selected-title']);
        });

        it('does not update title when selectedTitle is empty', async () => {
            const wrapper = createWrapper({
                selectedTitle: 'initial-title',
                modelValue: 'initial title'
            });

            await wrapper.setProps({selectedTitle: ''});
            await nextTick();

            expect(wrapper.emitted('update:modelValue')).toBeFalsy();
        });
    });
})
;
