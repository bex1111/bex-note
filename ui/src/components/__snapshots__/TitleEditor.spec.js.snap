// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TitleEditor.vue > matches snapshot 1`] = `
"<div class=\\"prime-fluid-stub\\" style=\\"margin-top: 0.75rem;\\">
  <div class=\\"prime-float-label-stub\\" variant=\\"in\\"><label for=\\"in_label\\">Give a title, like: Folder/Awesome title, Awesome title)</label>
    <prime-input-stub id=\\"in_label\\" modelvalue=\\"initial title\\" type=\\"text\\"></prime-input-stub>
  </div>
</div>"
`;
